File: examples/manager-integration.ts
import AdvancedTokenManager from '../src/index';

const auditLogger = {
    warn: (message: string) => console.log('[warn]', message),
    error: (message: string) => console.log('[error]', message)
};

const manager = new AdvancedTokenManager('manager-secret-change-me', ['salt-1', 'salt-2'], 'sha256', true, false, {
    logger: auditLogger,
    jwtDefaultAlgorithms: ['HS512'],
    throwOnValidationFailure: true,
    jwtMaxPayloadSize: 128,
    jwtAllowedClaims: ['transaction', 'amount']
});

const jwt = manager.generateJwt(
    { transaction: 'txn-123', amount: 99.99 },
    { algorithm: 'HS512', expiresIn: 120 }
);

const verified = manager.validateJwt(jwt);

console.log('Manager JWT:', jwt);
console.log('Manager payload:', verified);

try {
    manager.validateToken('not-a-valid-token');
} catch (error) {
    console.log('Token validation rejected input:', (error as Error).message);
}

try {
    const disallowedJwt = manager.generateJwt({ transaction: 'txn-999', scope: 'admin' });
    manager.validateJwt(disallowedJwt);
} catch (error) {
    console.log('Claim whitelist rejected JWT:', (error as Error).message);
}

File: examples/manager_integration.rs
use hash_token_rust::{
    AdvancedTokenManager, AdvancedTokenManagerOptions, Algorithm, JwtAlgorithm, JwtClaims,
    ManagerVerifyJwtOptions,
};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let manager = AdvancedTokenManager::new(
        Some("a-very-secure-secret-value".to_string()),
        Some(vec!["salt-a".into(), "salt-b".into()]),
        Some(Algorithm::Sha256),
        false,
        true,
        Some(AdvancedTokenManagerOptions {
            jwt_default_algorithms: Some(vec![JwtAlgorithm::HS256]),
            jwt_max_payload_size: Some(1024),
            ..Default::default()
        }),
    )?;

    let mut claims = JwtClaims::new();
    claims.insert("sub".to_string(), "user-123".into());

    let token = manager.generate_jwt(&claims, None)?;
    let verified: JwtClaims =
        manager.validate_jwt(&token, Some(ManagerVerifyJwtOptions::default()))?;

    println!("subject={}", verified["sub"]);
    Ok(())
}
